design-proposals/configurable-features.md

```diff
@@ -0,0 +1,77 @@
+# Overview
+With the introduction of [KubeVirt Feature Lifecycle](https://github.com/kubevirt/community/blob/main/design-proposals/feature-lifecycle.md) policy, features reaching General Availability (GA) need to drop their use of feature gates. This applies also to configurable features that we may want to disable.
+
+## Motivation
+Users or developers may want certain features to be in a given state, for example to make the best use out of given resources or for compliance reasons features may expose sensitive information from the host to the virtual machines (VM) or add additional containers to the launcher pod, which are not required by the user. The behavior of other features might be changed by editing configurables, e.g. the maximum of CPU sockets allowed for each VM can be configured.
+
+Before the introduction of [KubeVirt Feature Lifecycle](https://github.com/kubevirt/community/blob/main/design-proposals/feature-lifecycle.md) policy, many feature gates remained after feature's graduation to GA with the sole purpose of acting as a switch for the feature. Generally speaking, this is a bad practice, because feature gates should be reserved for controlling a feature until it reaches maturity. i.e., GA. Therefore, in the case that a developer wants to provide the ability to tune/change the state of the feature, configurables exposed in the KubeVirt CR should be provided. Ideally, this should be accomplished while achieving [eventually consistency](https://en.wikipedia.org/wiki/Eventual_consistency). This forces us to avoid the feature state control checking on webhooks and moving the feature state control closer to the responsible code. Moreover, it has to be decided how the system should behave if a virtual machine instance (VMi) is requiring a feature in a state different from what was configured in the KubeVirt CR, or what should happen if the configuration of a feature in use is changed. (see matrix below).
+
+## Goals
+- Establish how the features enablement switch should work.
+- Describe how the system should react in these scenarios:
+    - A feature in KubeVirt is set to state A and a VMi requests the feature to be in state B.
+    - A feature in KubeVirt is set to state A, there are running VMis using the feature in state A, and the feature is changed in KubeVirt to state B.
+    - A feature in KubeVirt is set to state A, and pending VMis want to use it.
+    - A feature in KubeVirt is set to state A, and running VMis using the feature in state B wants to live migrate.
+- Graduate as many features as possible from features gates to configurables.
```

Comment:
I think we should have a stronger statement in line with https://github.com/kubevirt/community/blob/main/design-proposals/feature-lifecycle.md:
After 3 releases in Beta, all feature gates must be dropped. All features that need configurables must have them by then.

Reply:
The intention of that goal is to move GA'd features to configurables or drop them.
In Section Update/Rollback compatibility, we state that before moving to GA, they should drop the feature gate and include a configurable if required. Does it adjust to what are you asking?

Reply:
I see. I've adjusted the goal and rewrote the Update/Rollback section to be a bit cleaner and to include your concern. Please, take a look.